lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -173,6 +173,7 @@ def after_harvest metadata
       # these until harvest before recording them
       def record_customer_metrics
         @counter_lock.synchronize do
+          return unless NewRelic::Agent.config[:'application_logging.metrics.enabled']
```

Comment:
I think we should return even earlier, and have this new `return` statement be the first line of the method (takes place before performing a lock synchronize operation.

Reply:
Great point! I'll adjust.